azure/scope/machine.go

```diff
@@ -101,59 +104,21 @@ type MachineScope struct {
 
 // MachineCache stores common machine information so we don't have to hit the API multiple times within the same reconcile loop.
 type MachineCache struct {
-	BootstrapData      string
+	BootstrapData      *string
 	VMImage            *infrav1.Image
-	VMSKU              resourceskus.SKU
-	availabilitySetSKU resourceskus.SKU
-}
-
-// InitMachineCache sets cached information about the machine to be used in the scope.
-func (m *MachineScope) InitMachineCache(ctx context.Context) error {
```

Comment:
@devigned do you have any concerns about replacing this "do a one-time initiatlization of all cached properties when we instantiate scope" vs implement the property value caching on demand?

My thinking (and this was @CecileRobertMichon 's thinking when we implemented caching in a prior PR) is that not all scope lifecycles are going to require access to all of these properties, and so in the act of caching them we are actually *de*optimizing things by fetching data unnecessarily.